### PR TITLE
[WEJBHTTP-14] Host header should include correct port information

### PR DESCRIPTION
**Pull request by spyrkob**
JIRA: https://issues.jboss.org/browse/WEJBHTTP-14

### CODE REVIEW

File: common/src/main/java/org/wildfly/httpclient/common/HttpTargetContext.java [hunk 1]
```diff
@@ -149,7 +149,7 @@ public void sendRequest(ClientRequest request, SSLContext sslContext, Authentica
     public void sendRequestInternal(final HttpConnectionPool.ConnectionHandle connection, ClientRequest request, AuthenticationConfiguration authenticationConfiguration, HttpMarshaller httpMarshaller, HttpResultHandler httpResultHandler, HttpFailureHandler failureHandler, ContentType expectedResponse, Runnable completedTask, boolean allowNoContent, boolean retry) {
         try {
             final boolean authAdded = retry || connection.getAuthenticationContext().prepareRequest(connection.getUri(), request, authenticationConfiguration);
-            request.getRequestHeaders().put(Headers.HOST, connection.getUri().getHost());
+            request.getRequestHeaders().put(Headers.HOST, connection.getUri().getHost() + ":" + connection.getUri().getPort());
```
Comment (rhatlapa) [inline common/src/main/java/org/wildfly/httpclient/common/HttpTargetContext.java]: Is it made sure that the port is defined in this case? I would say it is better to add port only if it is defined and leave it without it if it isn't.

**Comment by spyrkob** [comment]
@rhatlapa thanks, I updated the PR to check if the port was defined.

**Comment by rhatlapa** [comment]
Thanks @spyrkob Looks good to me now.

**Comment by stuartwdouglas** [comment]
If the port is 80 or 443 it should not be included, I will add an additional commit to fix

**Pull request merged**